drivers/hmis/app/models/hmis/ce/opportunity.rb

```diff
@@ -15,9 +15,9 @@ class Opportunity < GrdaWarehouseBase
                class_name: 'Hmis::WorkflowDefinition::Template'
 
     has_many :referrals, class_name: 'Hmis::Ce::Referral', dependent: :restrict_with_exception
-    has_many :categorizations, class_name: 'Hmis::Ce::OpportunityCategorization', foreign_key: :opportunity_id
+    has_many :categorizations, class_name: 'Hmis::Ce::OpportunityCategorization', foreign_key: :opportunity_id, dependent: :destroy
     has_many :categories, through: :categorizations
-    belongs_to :owner, polymorphic: true, optional: true # Hmis::Unit, ...
+    belongs_to :owner, polymorphic: true, optional: true # always an Hmis::Unit
```

Comment:
suggest simplifying the polymorphic relation if we are always referencing a unit